### DISCUSSION TITLE
Two domains on one host

### DISCUSSION THREAD

**Discussion by neuromobster** [community/community#41822] [Other Feature Feedback, Questions, & Ideas]
Hello everyone! Is it possible to have two domains on one repository? We bought two domains (.com & .org) and would like these 2 domains to point to the same web page as it would be inconvenient to update the content on both repositories every time. If not, is it otherwise possible to have somehow the .org domain redirect to the .com domain homepage without having two repositories? Thanks for reading.

**Comment by samharp** [comment]
Hello! Either way is possible, but I would highly recommend having one domain redirect to the other rather than keeping the same website live on both (even if you are sharing one repository). Keeping two websites live just puts duplicate data on your server, which can slow things down (largely depending on how big the website is). In terms of which domain to redirect to the other, that is personal preference and depends on how you want to set things up. Hope this helps!

**Reply by uuf6429** [reply]
> Either way is possible

Why are you telling him that? It's not possible and was never possible to have two (sub)domains handled by the same repository.

> keeping the same website live on both (even if you are sharing one repository)

You don't know that since GitHub does not publish how that part works.
In a typical apache or nginx site, for example, you set up the server config so that multiple (sub)domains point to the same document root (content/files). If GitHub somehow did end up supporting multiple (sub)domains, one expects they'd do it in this way as well.

**To be fair:** redirection is better for search engine reasons, since they don't like duplicate content.

**Reply by starding** [reply]
You are the right guy.

**Comment by DSDmark** [comment]
Hi @neuromobster,

It's worth noting that, regardless of which approach you choose, you will still need to maintain both domains separately, as they are treated as distinct entities by the domain name system. However, by using one of the above methods, you can ensure that both domains point to the same content, making it easier to update and manage your website.

**Comment by DSDmark** [comment]
## how To set two domains pointing to the same repository, you will need to follow these steps:

 -   Add the two domains to your repository settings. In the repository settings, under the "GitHub Pages" section, you will see a field labeled "Custom domain." You can enter your two domains here, separated by a comma.

  -  Configure your DNS settings to point both domains to your GitHub Pages site. To do this, you will need to add two "CNAME" records to your DNS configuration. The first record should point your first domain to your GitHub Pages site, and the second record should point your second domain to your GitHub Pages site.

  -  Wait for the DNS changes to propagate. It can take some time for the DNS changes to take effect, so you may need to wait a few hours or even a day for the domains to start pointing to your GitHub Pages site

**Reply by ivanmicetic** [reply]
The "Custom domain" field under "GitHub Pages" section of repository settings does not want to accept a second domain (comma separated). Is there a way to point a custom domain and another custom subdomain to the same GitHub pages repository?

**Reply by yacafx** [reply]
comma separated domain does not work :(

**Reply by okel17** [reply]
Is this something that will be supported in the future?

**Reply by moscowchill** [reply]
This would be nice.

**Reply by uuf6429** [reply]
A comment I saw recently by a GitHub EM seems to imply allowing only one custom domain is by design.

**Comment by Sandeepkasturi** [comment]
Absolutely it's possible to have multiple domains for one host, But i would recommend to redirect the sub domains to the main domain. 
But however you can have multiple domains, Here is my Screen Shot of having multiple domains provided and hosted on Vercel.com
![ss](https://github.com/community/community/assets/112337452/2c4c1214-ef10-47d9-928e-0e255bfb0ad2)

**Reply by moscowchill** [reply]
Redirecting without a web package is not always possible for all registrars.

**Reply by uuf6429** [reply]
@shorefall that's a screenshot of Vercel: https://www.prisma.io/docs/orm/more/help-and-troubleshooting/help-articles/vercel-caching-issue#vercel-uis-build-script-field

I think he misunderstood the question - since he's suggesting managing this stuff from Vercel side.
The right answer is NO.. GitHub does not allow *one repository* to handle *multiple (custom) domains*.

**Comment by Evidlo** [comment]
As others have mentioned, serving two domains from one repository does not seem to be possible.

Also I want to point out that if you attempt to do this by pointing two domains at Github pages, then adding two lines to your `CNAME` file, the second line will be ignored.  This means that any Github user can create their own repo with a `CNAME` matching the second line and effectively control your second domain!  Don't ask me how I know...
